### PR TITLE
added option without InsertAfter or InsertBefore and corrected some variable namings

### PR DESCRIPTION
**Pull request by outlawww**
added option to add rules if ACP is empty, therefore "InsertAfter" or "InsertBefore" can't be set.
Corrected the log options with correct variable names expected by "AccessRules" class.

### CODE REVIEW

**Review by amotolani — CHANGES_REQUESTED** [review]

**Comment by amotolani** [comment]
Just did some testing on this and it does work as it should but there are some other implications i noticed. Wondering if you did notice same while testing.

Testing with FMC API Version 6.3. 

When enabling logging (either `logBegin` or `logEnd` ) it seems the FMC API always expects that you set `sendEventsToFMC` to `true` . 

This is not the case while using the Management Console. In the Console you can set the logging destination to only `syslog` and that works fine. 

Here is a sample request body to the FMC API 

`
{
   "id":"005056BF-4E01-0ed3-0000-000268435459",
   "name":"Demo-Rule1903",
   "type":"AccessRule",
   "action":"ALLOW",
   "enabled":true,
   "sendEventsToFMC":false,
   "logBegin":true,
   "logEnd":false,
   "sourceNetworks":{
      "objects":[
         
      ],
      "literals":[
         {
            "type":"host",
            "value":"10.1.1.22"
         },
         {
            "type":"host",
            "value":"10.2.2.11"
         }
      ]
   },
   "destinationNetworks":{
      "objects":[
         
      ],
      "literals":[
         {
            "type":"host",
            "value":"10.4.4.88"
         }
      ]
   },
   "sourcePorts":{
      "objects":[
         {
            "name":"HTTPS",
            "id":"1834bd00-38bb-11e2-86aa-62f0c593a59a",
            "type":"ProtocolPortObject"
         }
      ]
   },
   "destinationPorts":{
      "objects":[
         {
            "name":"LDAP",
            "id":"1834d01a-38bb-11e2-86aa-62f0c593a59a",
            "type":"ProtocolPortObject"
         }
      ]
   },
   "enableSyslog":true,
   "newComments":[
      
   ],
   "commentHistoryList":[
      
   ]
}
`
and here is the API response 

`
{
  "error": {
    "category": "FRAMEWORK",
    "messages": [
      {
        "description": "Provide at least one destination for Connection Events to be logged to."
      }
    ],
    "severity": "ERROR"
  }
}
`
The implication of this is that the are errors when a user tries to do this same thing with the collection. Of course this is a an issue with the FMC API itself but it needs to be thought of here.

File: amotolani/cisco_fmc/plugins/modules/acp_rule.py [hunk 1]
```diff
@@ -1078,6 +1078,9 @@ def validate_net_obj_config(requested_config, config_name):
             obj1 = AccessRules(fmc=fmc1, acp_name=acp, action=action, name=name, section=section, insertBefore=insert_before)
         elif _create_obj and insert_after is not None:
             obj1 = AccessRules(fmc=fmc1, acp_name=acp, action=action, name=name, section=section, insertAfter=insert_after)
+        # Instantiate Access Rule with Section, without InsertAfter or InsertBefore
+        elif _create_obj:
```
[the remark this answers is not in the document]
Comment (outlawww) [reply amotolani/cisco_fmc/plugins/modules/acp_rule.py:1082]: I agree. But if we set the variables in ansible playbook to "null", will it still work? I' mean like:
```
      insert_after: null
      insert_before: null
```

**Comment by outlawww** [comment]
It seems they fixed the API in later releases, I just tested with FMC 6.6.1 and it works normal 
image

File: amotolani/cisco_fmc/plugins/modules/acp_rule.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (amotolani) [reply amotolani/cisco_fmc/plugins/modules/acp_rule.py:1082]: yes

[the remark this answers is not in the document]
Comment (amotolani) [reply amotolani/cisco_fmc/plugins/modules/acp_rule.py:1082]: Once we have this merged. I will push the new version to [galaxy](https://galaxy.ansible.com)

Thanks for contributing !

Comment (amotolani) [inline amotolani/cisco_fmc/plugins/modules/acp_rule.py:1082]:
```suggestion
        elif _create_obj and insert_before is None and insert_after is None:
```

Comment (amotolani) [reply amotolani/cisco_fmc/plugins/modules/acp_rule.py:1082]: I think it maybe a bit annoying that you have to pass `null` values for parameters you don't want to set anyway.

**Review by amotolani — APPROVED** [review]

**Pull request merged**